src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/ProjectDefaultConfigurationProvider.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Configuration
+{
+    /// <summary>
+    ///     Attempts to calculate the default configuration of a project without evaluating or loading any unneeded configurations.
+    /// </summary>
+    [Export(typeof(IProjectDefaultConfigurationProvider))]
+    [AppliesTo(ProjectCapabilities.ProjectConfigurationsDeclaredDimensions)]
+    [Order(Order.Default)]
+    internal class ProjectDefaultConfigurationProvider : IProjectDefaultConfigurationProvider
+    {
+        private readonly UnconfiguredProject _project;
+        private readonly IProjectThreadingService _threadingService;
+
+        [ImportingConstructor]
+        public ProjectDefaultConfigurationProvider(UnconfiguredProject project, IProjectThreadingService threadingService)
+        {
+            _project = project;
+            _threadingService = threadingService;
+
+            DimensionsProviders = new OrderPrecedenceImportCollection<IProjectConfigurationDimensionsProvider>(projectCapabilityCheckProvider: project);
+        }
+
+        [ImportMany]
+        public OrderPrecedenceImportCollection<IProjectConfigurationDimensionsProvider> DimensionsProviders
+        {
+            get;
+        }
+
+        public ProjectConfiguration GetDefaultConfiguration(IMinimalProjectConfiguration projectConfiguration)
+        {
+            return _threadingService.ExecuteSynchronously(() => GetDefaultConfigurationAsync(projectConfiguration));
```

Comment:
The dimension provider will need get a read lock to read project. I think this is unlikely introducing lots of delay, so the impact to the thread pool sounds limited.  if it turns out to impact performance, we can deal with it at that time.